### ISSUE TITLE
error when login is null

### ISSUE THREAD

**Issue by fsperling** [fpicalausa/remove-stale-branches#4]
Quite unsure what is happening.
Maybe the github token is missing permissions? Which ones are required?
Or something with the user that login is null?

``` /home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8928
                const { node: { branchName, prefix, target: { oid, author: { date, user: { login, organization }, }, }, }, } = ref;
                                                                                           ^

TypeError: Cannot read properties of null (reading 'login')
    at readBranches_1 (/home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8928:92)
    at readBranches_1.next (<anonymous>)
    at resume (/home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8842:44)
    at /home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8841:121
    at new Promise (<anonymous>)
    at Object.i.<computed> [as next] (/home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8841:63)
    at /home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:9074:152
    at Generator.next (<anonymous>)
    at fulfilled (/home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8955:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)```

**Comment by mhixon4479** [comment]
Hitting this as well - have you had any luck?

**Comment by fsperling** [comment]
I've added a workaround that skips the branch when user/login is null.
That way I can run it locally at least.

https://github.com/fsperling/remove-stale-branches/commit/e8e5ceb78de83df9aa47e53e2b4aec571fdcf3d5#diff-a5534a6e8574ae207e67d2e14977c9d085d7668680b9045d3db1f2a8927dd9a1R184-R187

**Comment by nurirppan** [comment]
> I've added a workaround that skips the branch when user/login is null. That way I can run it locally at least.
> 
> [fsperling@e8e5ceb#diff-a5534a6e8574ae207e67d2e14977c9d085d7668680b9045d3db1f2a8927dd9a1R184-R187](https://github.com/fsperling/remove-stale-branches/commit/e8e5ceb78de83df9aa47e53e2b4aec571fdcf3d5#diff-a5534a6e8574ae207e67d2e14977c9d085d7668680b9045d3db1f2a8927dd9a1R184-R187)

hi, i still got this error 

```
Run fpicalausa/remove-stale-branches@v1
  
Running in dry-run mode. No branch will be removed.
Looking at branch CICD/master
-> updated recently
Looking at branch Development/S02/0001/nama-user-story
-> last update was on 2021-08-24T03:43:07Z
-> marking as stale
/home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8928
                const { node: { branchName, prefix, target: { oid, author: { date, user: { login, organization }, }, }, }, } = ref;
                                                                                           ^

TypeError: Cannot read properties of null (reading 'login')
    at readBranches_1 (/home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8928:92)
    at readBranches_1.next (<anonymous>)
    at resume (/home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8842:44)
    at /home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8841:[12](https://github.com/pt-kompas-media-nusantara/kompas-id-ios/actions/runs/7285098053/job/19851461918#step:3:13)1
    at new Promise (<anonymous>)
    at Object.i.<computed> [as next] (/home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8841:63)
    at /home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:9074:[15](https://github.com/pt-kompas-media-nusantara/kompas-id-ios/actions/runs/7285098053/job/19851461918#step:3:16)2
    at Generator.next (<anonymous>)
    at fulfilled (/home/runner/work/_actions/fpicalausa/remove-stale-branches/v1/dist/index.js:8955:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

this is my code
```
name: Remove Stale Branches

on:
  push:
  # schedule:
  #   - cron: "0 0 * * *" # Everday at midnight

jobs:
  remove-stale-branches:
    name: Remove Stale Branches
    runs-on: ubuntu-latest

    steps:
      - name: Checkout Repository
        uses: actions/checkout@v4
        with:
          fetch-depth: 0

      - uses: fpicalausa/remove-stale-branches@v1
        with:
          dry-run: true # Check out the console output before setting this to false
```

**Comment by fpicalausa** [comment]
Thanks for the report and the proposed fix @fsperling. The latest version of the action shouldn't have been destructuring gql results as aggressively (and I wasn't able to reproduce the specific issue), but the graphql type exposed by github shows that author and users can be null. There is likely a case where missing privileges can cause this, but also when a committer isn't a github user (e.g. if the committer info is configured differently from the github user locally)

#5 solves this by (1) correctly handling empty/null values; and (2) giving you the option to `ignore-unknown-authors`, along with setting a default recipient to be notified when branches will be deleted. This is published as version 1.6.0, so be sure to update.
